### PR TITLE
added ner.md #hacktoberfest

### PR DESCRIPTION
**Pull request by UmaGunturi**
This PR includes:
- a brief introduction to NER(Named Entity Recognition)
- approaches and use cases of NER
- few tools/resources for NER

### CODE REVIEW

**Comment by neomatrix369** [comment]
Thanks for this @UmaGunturi I will review this soon and get back to you

File: natural-language-processing/ner.md [hunk 1]
```diff
@@ -1,26 +1,95 @@
...
+
+The ACE challenges use a more complex evaluation metric which includes a weighting schema, Check References for deeper understanding.
+Replicating experiments and baselines from ACE are a little complex since all the datasets and results are not open and free, so I guess this challenge results and experiments will fade away with time.  
+
+**3.4 Message Understanding Conference (MUC)**  
+
+MUC introduced detailed metrics in an evaluation considering different categories of errors these metrics can be defined as in terms of comparing the response of a model against golden annotation:  
+
+* **Correct (COR):** both are the same;   
+* **Incorrect (INC):** the output of a system and the golden annotation don’t match;
```
Comment (neomatrix369) [inline natural-language-processing/ner.md:63]: > golden annotation
Is this same as ground truth?

File: natural-language-processing/ner.md [hunk 2]
```diff
@@ -1,26 +1,95 @@
...
+**3.4 Message Understanding Conference (MUC)**  
+
+MUC introduced detailed metrics in an evaluation considering different categories of errors these metrics can be defined as in terms of comparing the response of a model against golden annotation:  
+
+* **Correct (COR):** both are the same;   
+* **Incorrect (INC):** the output of a system and the golden annotation don’t match;    
+* **Partial (PAR):** system and the golden annotation are somewhat “similar” but not the same;   
+* **Missing (MIS):** a golden annotation is not captured by a system;  
+* **Spurius (SPU):** model produces a response which doesn’t exist in the golden annotation;  
+
```
Comment (neomatrix369) [inline natural-language-processing/ner.md:67]: **MIS** and **SPU** seem to be similar to me here - any thoughts?

File: natural-language-processing/ner.md [hunk 3]
```diff
@@ -1,26 +1,95 @@
...
+**4.3 Customer Support**  
+
+There are a number of ways to make the process of customer feedback procedure smooth and easy by using NER. Let’s take an example to understand the process. If you are handling the customer support department of an electronic store with multiple branches worldwide, you go through a number mentions in your customers’ feedback. 
+
+**5 Conclusion**  
+
+Named Entity Recognition has been developing continuously for over 15 years. The novel use is to extract different types of information (name, date, time, location) from the text. In addition, there are more than 20 languages and more than 200 types of entities. Most researches are interested in specific information on topic types such as news articles, web page information, etc..This provides an overview of techniques for creating a NERC system, from manual rule-based assignments to providing good and accurate results. But it comes with the time it takes to set the rules like, Supervised learning requires a large corpus that has been labeled, Semi-supervised and unsupervised learning allows for rapid recognition of entities without having to have a large labeled corpus.
```
Comment (neomatrix369) [inline natural-language-processing/ner.md:86]: > etc..This

would be

>  etc... This

File: natural-language-processing/ner.md [hunk 4]
```diff
@@ -1,26 +1,95 @@
...
 
+**References:**  
+📌 To know more about NER Deep learning approaches - Refer to https://www.depends-on-the-definition.com/introduction-named-entity-recognition-python/  
+📌 More details about LSTMs can be found [here](http://colah.github.io/posts/2015-08-Understanding-LSTMs/)
+📌 An implementation of the MUC evaluation metrics can be found:
+* https://github.com/jantrienes/nereval
+* [“Automatic Content Extraction 2008 Evaluation Plan (ACE08)”](http://www.eng.utah.edu/~cs6961/papers/ACE-2008-description.pdf)
+* [“The Automatic Content Extraction (ACE) Program Tasks, Data, and Evaluation”](https://pdfs.semanticscholar.org/0617/dd6924df7a3491c299772b70e90507b195dc.pdf) 
+* http://www.aclweb.org/anthology/M93-1007
```
Comment (neomatrix369) [inline natural-language-processing/ner.md:95]: _Suggestions:_
Can you please go thru the resources (and also presentations) on https://bit.ly/better-nlp-launch and see if there are any useful resources you could pick and add to the above list.

Also one of the presentations slides list a number of components to NER via either Spacey or NLTK - would be good to include that list above somewhere.

Thanks

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
LGTM 

Overall looks very good, can you please address all the comments (whichever need fixing) and also attempt to resolve other comments (requests and suggestions -- if possible)

Thanks again @UmaGunturi

File: natural-language-processing/course-tutorial-learning-resources.md [hunk 5]
```diff
@@ -7,6 +7,7 @@
 - [How to solve 90% of NLP problems: a step-by-step guide](https://blog.insightdatascience.com/how-to-solve-90-of-nlp-problems-a-step-by-step-guide-fda605278e4e)
 - [A framework for dialog research + datasets](https://parl.ai/)
 - [How Rossum is using deep learning to extract data from any document](https://www.linkedin.com/posts/eric-feuilleaubois-ph-d-43ab0925_how-rossum-is-using-deep-learning-to-extract-activity-6605832802078347264-ZsW8)
+- [Everything you need to know about Named Entity Recognition!!](https://github.com/UmaGunturi/awesome-ai-ml-dl/blob/master/natural-language-processing/ner.md)
```
[the remark this answers is not in the document]
Comment (UmaGunturi) [reply natural-language-processing/course-tutorial-learning-resources.md]: Done!

**Review by UmaGunturi — COMMENTED** [review]
Made few changes requested!

File: natural-language-processing/ner.md (hunk 4 above)
Comment (UmaGunturi) [reply natural-language-processing/ner.md:91]: Sure! Will check and update.

Comment (neomatrix369) [reply natural-language-processing/ner.md:91]: Maybe via a new PR, let's merge this for now

File: natural-language-processing/ner.md [hunk 6]
```diff
@@ -1,26 +1,91 @@
...
+
+The ACE challenges use a more complex evaluation metric which includes a weighting schema, Check References for deeper understanding.
+Replicating experiments and baselines from ACE are a little complex since all the datasets and results are not open and free, so I guess this challenge results and experiments will fade away with time.  
+
+**3.4 Message Understanding Conference (MUC)**  
+
+The MUC-5 Scoring System is evaluation software that aligns and scores the templates produced by the information extraction systems under evaluation in comparison to an "answer key" created by humans . The Scoring System produces comprehensive summary reports showing the overall scores for the templates in the test set ; these may be supplemented by detailed score reports showing scores for each template individually. To understand the scoring categories, see [this](https://www.aclweb.org/anthology/M93-1007.pdf)
+
+Below is a summary table with meaning of each scoring category!
+![scoring](https://github.com/UmaGunturi/awesome-ai-ml-dl/blob/master/natural-language-processing/formulae/scoring.png)
```
Comment (neomatrix369) [inline natural-language-processing/ner.md:63]: Url is still pointing to https://github.com/UmaGunturi -  this is a simple change I think

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
Last couple of changes I think

Otherwise we can go ahead and merge it

File: natural-language-processing/ner.md (hunk 6 above)
Comment (UmaGunturi) [reply natural-language-processing/ner.md]: Done!

**Comment by UmaGunturi** [comment]
> Last couple of changes I think
> 
> Otherwise we can go ahead and merge it

Done, please review!

**Review by neomatrix369 — APPROVED** [review]
LGTM merging now

**Pull request merged**